### PR TITLE
[CLI][Bundle][Packager] Implement --root --assetRoot and --out for consistency with packager and more flexibility

### PR DESCRIPTION
**Pull request by JohnyDays**
Needed this to bundle my non-standard react-native app.

### CODE REVIEW

**Comment by brentvatne** [comment]
Seems reasonable - right now my workaround is to create a `index.ios.js` that requires my app, but this seems cleaner.

**Comment by JohnyDays** [comment]
Also doesn't force you to have an ````iOS```` folder in your project, wouldn't be too bad to just create one, but it shouldn't be required imo.

**Comment by jessepollak** [comment]
:+1:

**Comment by MartinMoizard** [comment]
:+1: thanks !

**Comment by amasad** [comment]
looks good, thanks!

File: local-cli/bundle.js [hunk 1]
```diff
@@ -9,12 +9,25 @@ var OUT_PATH = 'iOS/main.jsbundle';
 
 function getBundle(flags) {
 
+  var outPath = flags.out ? flags.out : OUT_PATH;
+
+  var projectRoots = [path.resolve(__dirname, '../../..')];
+  if(flags.root){
```
Comment (vjeux) [inline local-cli/bundle.js]: nit: please put space after `if` and before `{`

```
if (flags.root) {
```

File: local-cli/bundle.js [hunk 2]
```diff
@@ -9,12 +9,25 @@ var OUT_PATH = 'iOS/main.jsbundle';
...
+    projectRoots.push(path.resolve(flags.root));
+  }
+
+  var assetRoots = [path.resolve(__dirname, '../../..')];
+  if(flags.assetRoot){
```
Comment (vjeux) [inline local-cli/bundle.js]: same here

File: local-cli/bundle.js [hunk 3]
```diff
@@ -9,12 +9,25 @@ var OUT_PATH = 'iOS/main.jsbundle';
...
+
   var options = {
-    projectRoots: [path.resolve(__dirname, '../../..')],
+    projectRoots: projectRoots,
     transformModulePath: require.resolve('../packager/transformer.js'),
-    assetRoots: [path.resolve(__dirname, '../../..')],
+    assetRoots: assetRoots,
     cacheVersion: '2',
-    blacklistRE: blacklist('ios')
+    blacklistRE: blacklist('ios'),
```
Comment (vjeux) [inline local-cli/bundle.js]: yay trailing commas! :+1:

**Comment by JohnyDays** [comment]
Done @vjeux, thanks! :+1:

**Comment by JohnyDays** [comment]
Sorry @vjeux, anything left for this to be merged? It's getting old to manually modify this file every time I run npm install. Thanks!

**Comment by vjeux** [comment]
@amasad imported your code, should land today and be in the repo tomorrow

**Comment by JohnyDays** [comment]
Thanks!

**Comment by medington** [comment]
:+1: @JohnyDays - Needed this functionality for out build process as well

**Comment by medington** [comment]
@vjeux Can you please review the comments above regarding my concerns on the casing of --assetRoot vs. --assetRoot**s** (plural) before this PR is closed? As stated above, my vote is for consistency.

**Comment by JohnyDays** [comment]
@medington Might as well implement it, did it in fcc4e4f

**Comment by medington** [comment]
@JohnyDays did you see the error from the Travis log:

/private/tmp/react-native-VNYW6hxt/EndToEndTest/node_modules/react-native/local-cli/bundle.js:23
    });
      ^
**SyntaxError: Unexpected token ;**

Isn't CI a wonderful thing? ;-)

**Comment by JohnyDays** [comment]
It's what I get for editing in github's editor :cry:   oops

**Comment by vjeux** [comment]
I'm sorry but i was in pto for the last week. @amasad, can you review this pull request?

**Comment by amasad** [comment]
OK updated the internal diff with the recent changes

**Comment by MattFoley** [comment]
:+1:

**Comment by MattFoley** [comment]
Are there still plans to merge this in?

**Comment by amasad** [comment]
yes

**Pull request merged**